core/src/main/java/org/apache/calcite/jdbc/CalciteMetaImpl.java

```diff
@@ -178,11 +205,16 @@ private <E> MetaResultSet createResultSet(Enumerable<E> enumerable,
     for (String name : names) {
       final int index = fields.size();
       final String fieldName = AvaticaUtils.toCamelCase(name);
-      final Field field;
+      Field field;
       try {
         field = clazz.getField(fieldName);
       } catch (NoSuchFieldException e) {
-        throw new RuntimeException(e);
+        try {
+          // Check if subclass contains the desired field.
+          field = clazz.getDeclaredField(fieldName);
+        } catch (NoSuchFieldException e2) {
+          throw new RuntimeException(e2);
+        }
       }
```

Comment:
I am surprised that `getDeclaredField` would ever succeed if `getField` has failed.

Reply:
I was testing on my subclasses and `getField()` returns the 10 fields of the default `MetaTable`  class attributes, and `getDeclaredField()` returns the additional properties on my subclass.